src/main/scala/java/time/Instant.scala

```diff
@@ -0,0 +1,267 @@
+  // def range(field: TemporalField): ValueRange
+
+  // Implemented by TemporalAccessor
+  // def get(field: TemporalField): Int
+
+  def getLong(field: TemporalField): Long = field match {
+    case INSTANT_SECONDS => seconds
+    case NANO_OF_SECOND  => nanos
+    case MICRO_OF_SECOND => nanos / NANOS_IN_MICRO
+    case MILLI_OF_SECOND => nanos / NANOS_IN_MILLI
+
+    case _: ChronoField =>
+      throw new UnsupportedTemporalTypeException(s"Field not supported: $field")
+
+    case _ => field.getFrom(this)
+  }
+
+  def getEpochSecond(): Long = seconds
+
+  def getNano(): Int = nanos
+
+  def `with`(field: TemporalField, value: Long): Instant = {
+    val msg = s"Invalid value for field $field: $value"
+    field match {
+      case INSTANT_SECONDS =>
+        requireDateTime(value >= MinSecond && value <= MaxSecond, msg)
+        if (value == seconds) this
+        else ofEpochSecond(value, nanos)
+
+      case NANO_OF_SECOND =>
+        requireDateTime(value >= 0 && value <= MaxNanosInSecond, msg)
+        if (value == nanos) this
+        else ofEpochSecond(seconds, value)
+
+      case MICRO_OF_SECOND =>
+        requireDateTime(value >= 0 && value <= MaxNanosInSecond / NANOS_IN_MICRO, msg)
+        val newNanos = value * NANOS_IN_MICRO
+        if (newNanos == nanos) this
+        else ofEpochSecond(seconds, newNanos)
+
+      case MILLI_OF_SECOND =>
+        requireDateTime(value >= 0 && value <= MaxNanosInSecond / NANOS_IN_MILLI, msg)
+        val newNanos = value * NANOS_IN_MILLI
+        if (newNanos == nanos) this
+        else ofEpochSecond(seconds, newNanos)
+
+      case _: ChronoField =>
+        throw new UnsupportedTemporalTypeException(s"Field not supported: $field")
+
+      case _ => field.adjustInto(this, value)
+    }
+  }
+
+  def truncatedTo(unit: TemporalUnit): Instant = {
+    if (unit == NANOS) this
+    else {
+      val duration = unit.getDuration
+      if (duration.getSeconds > SECONDS_IN_DAY) {
+        throw new UnsupportedTemporalTypeException("Unit too large")
+      }
+
+      val unitNanos = duration.toNanos
+      if ((unitNanos % NANOS_IN_DAY) != 0) {
+        throw new UnsupportedTemporalTypeException("Unit must be a multiple of a standard day")
+      }
+
+      val newTotalNanos = totalNanos / unitNanos * unitNanos
+      val newSeconds = Math.floorDiv(newTotalNanos, NANOS_IN_SECOND)
+      val newNanos = Math.floorMod(newTotalNanos, NANOS_IN_SECOND)
+      ofEpochSecond(newSeconds, newNanos)
+    }
+  }
+
+  def plus(amount: Long, unit: TemporalUnit): Instant = unit match {
+    case NANOS     => plusNanos(amount)
+    case MICROS    => plusNanos(amount * NANOS_IN_MICRO)
+    case MILLIS    => plusMillis(amount)
+    case SECONDS   => plusSeconds(amount)
+    case MINUTES   => plusSeconds(amount * SECONDS_IN_MINUTE)
+    case HOURS     => plusSeconds(amount * SECONDS_IN_HOUR)
+    case HALF_DAYS => plusSeconds((amount * SECONDS_IN_DAY) / 2)
+    case DAYS      => plusSeconds(amount * SECONDS_IN_DAY)
```

Comment:
These `*` may overflow as there are no checks on the value of `amount`. Consider using `Math.multiplyExact`.